src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlStoreClient.cs

```diff
@@ -243,9 +245,20 @@ internal async Task<long> MergeResourcesGetTransactionVisibilityAsync(Cancellati
                 }
                 catch (Exception e)
                 {
+                    var sqlEx = e as SqlException;
+                    if (sqlEx != null && sqlEx.Number == SqlStoreErrorCodes.MergeResourcesConcurrentCallsIsAboveOptimal)
+                    {
+                        _logger.LogWarning(e, $"Error on {nameof(MergeResourcesBeginTransactionAsync)}: MergeResources concurrent calls is above optimal. Delay={delayOnOverloadMilliseconds} milliseconds.");
+
+                        // TODO: Prepare to throw 429 instead of wait/delay when bundle code is ready
+                        await Task.Delay(delayOnOverloadMilliseconds, cancellationToken);
```

Comment:
Should we add a max number of retries, similar to the timeouts? 
Or should that be included as part of the bundle handling part of the code?

Reply:
I don't think so. Please look at my answer to RB's comment.